src/Orchard.Environment.Extensions/Compilers/CSharpExtensionCompiler.cs

```diff
@@ -267,10 +555,38 @@ private static string GetLanguageVersion(string languageVersion)
             return languageVersion;
         }
 
-        private static Command RunCsc(string[] cscArgs)
+        private bool CheckMissingIO(IEnumerable<string> inputs, IEnumerable<string> outputs)
+        {
+            if (!inputs.Any() || !outputs.Any())
+            {
+                return false;
+            }
+
+            return CheckMissingIO(inputs) || CheckMissingIO(outputs);
+        }
+
+        private bool CheckMissingIO(IEnumerable<string> items)
         {
-            // Locate CoreRun
-            return Command.Create("csc.dll", cscArgs);
+            return items.Where(i => !File.Exists(i)).Any();
+        }
+
+        private bool TimestampsChanged(IEnumerable<string> inputs, IEnumerable<string> outputs)
+        {
+            // Find the output with the earliest write time
+            var minDateUtc = DateTime.MaxValue;
+
+            foreach (var outputPath in outputs)
+            {
+                var lastWriteTimeUtc = File.GetLastWriteTimeUtc(outputPath);
+
+                if (lastWriteTimeUtc < minDateUtc)
+                {
+                    minDateUtc = lastWriteTimeUtc;
+                }
+            }
```

Comment:
replace this with....

var minDateUtc = outputs.Min(output => File.GetLastWriteTimeUtc(output));

maybe?